accelerometer/circadianRhythms.py

```diff
@@ -0,0 +1,125 @@
+    :param int epochPeriod: Size of epoch time window (in seconds)
+    :param bool fourierWithAcc:True calculates fourier done with acceleration data instead of sleep data
+    :param list(str) labels: Activity state labels
+    :param dict summary: Output dictionary containing all summary metrics
+
+    :return: Write dict <summary> keys 'PSD-<W/Hz>'
+    """
+    if fourierWithAcc:
+        y = (e['accImputed'])
+    else:
+        cols = []
+        # get imputed variable names for each activity type
+        # NB - make sure sleep comes first
+        for accType in labels:
+            col = accType + 'Imputed'
+            cols += [col]
+      
+        idx = cols.index('sleepImputed') #gets index of sleep label
+        # collects the sleep column from data frame assumes sleep if it is the highest imputed value
+        y = (np.argmax(e[cols].values, axis=1) == idx).astype('int')*2-1
+        
+    n = len(y)
+    k = len(y)*epochPeriod/(60*60*24)
+    e = -2.j * np.pi * k * np.arange(n) / n
+    # finds the power spectral density for a one day cycle using fourier analysis 
+    res = np.sum(np.exp(e) * y, axis=-1)/n
+    PSD = np.abs(res)**2
+    summary['PSD'] = PSD
+
+def calculateFourierFreq(e, epochPeriod, fourierWithAcc, labels, summary):
+    """Calculate the most prevalent frequency in a fourier analysis 
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param int epochPeriod: Size of epoch time window (in seconds)
+    :paran bool fourierWithAcc: True calculates fourier done with acceleration data instead of sleep data
+    :param list(str) labels: Activity state labels
+    :param dict summary: Output dictionary containing all summary metrics
+
+    :return: Write dict <summary> keys 'fourier frequency-<1/days>'
+    """
+    if fourierWithAcc:
+        y = (e['accImputed'])
+    else:
+        cols = []
+        # get imputed variable names for each activity type
+        # NB - make sure sleep comes first
+        for accType in labels:
+            col = accType + 'Imputed'
+            cols += [col]
+            
+        idx = cols.index('sleepImputed') #gets index of sleep label
+        # collects the sleep column from data frame assumes sleep if it is the highest imputed value
+        y = (np.argmax(e[cols].values, axis=1) == idx).astype('int')*2-1
+        
+    # fast fourier transform of the sleep column 
+    fft_y = np.abs(fftpack.fft(y))
+    
+    i =  np.arange(1,len(fft_y)) 
+    k_max = np.argmax(fft_y[i]) + 1
+    n = len(y)
+    # maximises the fourier transform function (func) using the fft_y as a first esitmate 
+    func = lambda k: -np.abs(np.sum(np.exp(-2.j * np.pi * k * np.arange(n) / n) * y, axis=-1)/n)
+    res = sp.optimize.minimize_scalar(func, bracket = (k_max-1,k_max+1)) 
+    #adjusts the frequency to have the units 1/days
+    freq_mx = float(res.x)/(len(y)*epochPeriod/(60*60*24))
+    summary['fourier-frequency'] = freq_mx
+    
```

Comment:
+ 3 space gap

Reply:
fixed in ce9842906efbd024c24d668b2d3bd0b3f429f559